apps/frontend/core-dapp/src/components/SettingsMenu/WalletIdentity.tsx

```diff
@@ -0,0 +1,93 @@
+import { FC, Suspense, useMemo } from 'react'
+import { Flex } from 'prepo-ui'
+import { suspend } from 'suspend-react'
+import { createFallthroughProvider } from 'prepo-stores'
+import { NETWORKS } from 'prepo-constants'
+import { observer } from 'mobx-react-lite'
+import { TestIds } from '../TestId'
+import WalletAvatar from '../../features/connect/WalletAvatar'
+import { getShortAccount } from '../../utils/account-utils'
+import { useRootStore } from '../../context/RootStoreProvider'
+
+type Props = {
+  address: string
+}
+
+type SID = {
+  getName: (address: string) => Promise<{ name: string | null }>
+}
+
+const useSpaceId = ({
+  address,
+}: {
+  address: string
+}): {
+  dotArbName?: string
+  dotEthName?: string
+} => {
+  const {
+    web3Store: { coreProvider },
+  } = useRootStore()
+
+  const arbitrumProvider = useMemo(
+    () =>
+      coreProvider.network.chainId === NETWORKS.arbitrumOne.chainId
+        ? coreProvider
+        : createFallthroughProvider(NETWORKS.arbitrumOne),
+    [coreProvider]
+  )
+
+  return suspend(async () => {
+    // @ts-expect-error - Library does not provide types
+    const { default: SID, getSidAddress } = await import('@siddomains/sidjs')
+
+    const arbId: SID = new SID({
+      provider: arbitrumProvider,
+      sidAddress: getSidAddress(NETWORKS.arbitrumOne.chainId),
+    })
+
+    const ethProvider = createFallthroughProvider(NETWORKS.mainnet)
+    const ethId: SID = new SID({
+      provider: ethProvider,
+      sidAddress: getSidAddress(NETWORKS.mainnet.chainId),
+    })
+
+    const [{ name: dotArbName }, { name: dotEthName }] = await Promise.all([
+      arbId.getName(address),
+      ethId.getName(address),
+    ])
+
+    if (dotArbName) return { dotArbName }
+    return { dotEthName: dotEthName ?? undefined }
```

Comment:
Wouldn't the empty value of dotArbName and dotEthName be undefined?
```suggestion
    return { dotArbName, dotEthName }
```

Reply:
No, because the Space ID SDK returns null if they can't find an ID, so I'd rather not mix null and undefined